src/dtx/dtx_common.c

```diff
@@ -169,6 +138,15 @@ dtx_free_dbca(struct dtx_batched_cont_args *dbca)
 		D_FREE(dbpa);
 	}
 
+	if (dbca->dbca_cleanup_req != NULL)
+		sched_req_put(dbca->dbca_cleanup_req);
+
+	if (dbca->dbca_commit_req != NULL)
+		sched_req_put(dbca->dbca_commit_req);
+
+	if (dbca->dbca_agg_req != NULL)
+		sched_req_put(dbca->dbca_agg_req);
```

Comment:
I can't think of how can above "xxx_req != NULL", shouldn't we assert "xxx_req == NULL"?

Reply:
Because the sponsor of related XXX_req may has exited without cleanup XXX_req.